Modern/utilities/src/main/java/org/bonej/utilities/ElementUtil.java

```diff
@@ -119,16 +116,33 @@ private ElementUtil() {}
 			return true;
 		}
 
-		final Set<Double> colors = new HashSet<>();
+		//a and b have the first pixel value
+		double a = interval.firstElement().getRealDouble();
+		double b = a;
+		double c;
+		
 		final Cursor<T> cursor = interval.cursor();
-		while (cursor.hasNext()) {
+		while (cursor.hasNext()){
 			cursor.fwd();
-			final T e = cursor.get();
-			colors.add(e.getRealDouble());
-			if (colors.size() > 2) {
-				return false;
+			c = cursor.get().getRealDouble();
+			//if we encounter a different pixel value
+			// than a, assign it to b, and stop
+			if (c != a) {
+				b = c;
+				break;
```

Comment:
Could have just one loop with

```java
double colors = 1;
while (cursor.hasNext()) {
    c = cursor.get().getRealDouble();
    if (c == a) {
        continue;
    }
    if (colors == 1) {
         b = c;
         colors++;
    } else if (c != b) {
        return false;
    }
}
```

Reply:
true, but it means that for many if not most of the pixels you have to do an extra if, assignment and increment, unnecessarily.